operation/fhir-operation-reindex/src/main/java/com/ibm/fhir/operation/reindex/ReindexOperation.java

```diff
@@ -0,0 +1,149 @@
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+import java.util.List;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.exception.FHIROperationException;
+import com.ibm.fhir.model.format.Format;
+import com.ibm.fhir.model.parser.FHIRParser;
+import com.ibm.fhir.model.resource.OperationDefinition;
+import com.ibm.fhir.model.resource.OperationOutcome;
+import com.ibm.fhir.model.resource.OperationOutcome.Issue;
+import com.ibm.fhir.model.resource.Parameters;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.type.code.IssueSeverity;
+import com.ibm.fhir.model.type.code.IssueType;
+import com.ibm.fhir.server.operation.spi.AbstractOperation;
+import com.ibm.fhir.server.operation.spi.FHIROperationContext;
+import com.ibm.fhir.server.operation.spi.FHIRResourceHelpers;
+import com.ibm.fhir.server.util.FHIROperationUtil;
+
+/**
+ * Custom operation to invoke the persistence layer reindexing of resources
+ */
+public class ReindexOperation extends AbstractOperation {
+    private static final Logger logger = Logger.getLogger(ReindexOperation.class.getName());
+    
+    private static final String PARAM_TSTAMP = "_tstamp";
+    private static final String PARAM_RESOURCE_COUNT = "_resourceCount";
+    
+    static final DateTimeFormatter DAY_FORMAT = new DateTimeFormatterBuilder()
+            .appendPattern("yyyy-MM-dd")
+            .parseDefaulting(ChronoField.NANO_OF_DAY, 0)
+            .toFormatter()
+            .withZone(ZoneId.of("UTC"));
+    
+    public ReindexOperation() {
+        super();
+    }
+
+    @Override
+    protected OperationDefinition buildOperationDefinition() {
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream("reindex.json")) {
+            return FHIRParser.parser(Format.JSON).parse(in);
+        } catch (Exception e) {
+            throw new Error(e);
+        }
+    }
+
+    @Override
+    protected Parameters doInvoke(FHIROperationContext operationContext, Class<? extends Resource> resourceType,
+            String logicalId, String versionId, Parameters parameters, FHIRResourceHelpers resourceHelper)
+            throws FHIROperationException {
+        
+        // Allow only POST because we're changing the state of the database
+        String method = (String) operationContext.getProperty(FHIROperationContext.PROPNAME_METHOD_TYPE);
+        if (!"POST".equalsIgnoreCase(method)) {
+            throw new FHIROperationException("HTTP method not supported: " + method);
+        }
+        
+        try {
+            Instant tstamp = Instant.now();
+            int resourceCount = 10;
+            
+            if (parameters != null) {
+                for (Parameters.Parameter parameter : parameters.getParameter()) {
+                    if (parameter.getValue() != null) {
+                        logger.fine("reindex param: " + parameter.getName().getValue() + " = " + parameter.getValue().toString());
```

Comment:
since the design of $reindex involves getting invoked a lot in parallel, its probably worth guarding this with a logger.isLoggable

Reply:
agreed

Reply:
Is this updated?